api/v1alpha1/hostedcluster_types.go

```diff
@@ -42,10 +42,65 @@ type HostedClusterSpec struct {
 
 	SSHKey corev1.LocalObjectReference `json:"sshKey"`
 
+	// ProviderCreds is a reference to a secret containing cloud account info
 	ProviderCreds corev1.LocalObjectReference `json:"providerCreds"`
 
+	// Networking contains network-specific settings for this cluster
+	Networking ClusterNetworking `json:"networking"`
+
+	Platform PlatformSpec `json:"platform"`
+
+	// InfraID is used to identify the cluster in cloud platforms
+	InfraID string `json:"infraID,omitempty"`
+}
+
+type ClusterNetworking struct {
 	ServiceCIDR string `json:"serviceCIDR"`
 	PodCIDR     string `json:"podCIDR"`
+	MachineCIDR string `json:"machineCIDR"`
+}
+
+type PlatformSpec struct {
+	// AWS contains AWS-specific settings for the HostedCluster
+	// +optional
+	AWS *AWSPlatformSpec `json:"aws,omitempty"`
+}
+
+type AWSPlatformSpec struct {
+	// Region is the AWS region for the cluster
+	Region string `json:"region"`
+
+	// AvailabilityZone is the default availability zone for the cluster
+	// +optional
+	AvailabilityZone string `json:"availabilityZone,omitempty"`
```

Comment:
have we considered  `Region` and `AvailabilityZone` as part of `NodePoolDefaults`?

Reply:
@enxebre I don't think we will support multi-region clusters, so I think of Region as a cluster-level thing. 
AvailabilityZone yes is more of a nodepool level thing. However, the cluster cloud config (in the control plane) forces me to specify a zone (even though all it does with the zone is derive a region from it). I can definitely move it to the nodepool default and get the default from there for control plane purposes.